fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryPlan.java

```diff
@@ -88,68 +94,16 @@
-     * natural order of results of this plan. The return value is <code>true</code> if the plan
-     * returns elements in descending order and <code>false</code> if the elements are
-     * returned in ascending order.
-     * @return <code>true</code> if this plan returns elements in reverse order.
-     */
-    boolean isReverse();
-
-    /**
-     * Indicates whether this plan (or one of its components) scans at least a subset of the record range directly
-     * rather than going through a secondary index.
-     * A plan may only scan over a subset of all records. Someone might not use any secondary indexes explicitly, but
-     * they might make use of the primary key index. For example, if they had a compound primary key, and they issued
-     * a query for all records that had some value for the first element of their primary key, the planner will produce
-     * a plan which scans over a subset of all records.
-     * @return <code>true</code> if this plan (or one of its components) scans at least a subset of the records directly.
-     */
-    boolean hasRecordScan();
-
-    /**
-     * Indicates whether this plan (or one of its components) must perform a scan over all records in the store directly
-     * rather than going through a secondary index.
-     * See {@link #hasRecordScan hasRecordScan} for the comparison between two methods.
-     * @return <code>true</code> if this plan (or one of its components) must perform a scan over all records in the
-     * store directly.
-     */
-    boolean hasFullRecordScan();
-
-    /**
-     * Indicates whether this plan scans the given index.
-     * @param indexName the name of the index to check for
-     * @return <code>true</code> if this plan (or one of its children) scans the given index.
-     */
-    boolean hasIndexScan(@Nonnull String indexName);
-
-    /**
-     * Returns a set of names of the indexes used by this plan (and its sub-plans).
-     * @return an set of indexes used by this plan.
-     */
-    @Nonnull
-    Set<String> getUsedIndexes();
-
-    /**
-     * Adds one to an appropriate {@link StoreTimer} counter for each plan and subplan of this plan, allowing tracking
-     * of which plans are being chosen (e.g. index scan vs. full scan).
-     * @param timer the counters to increment
-     */
-    void logPlanStructure(StoreTimer timer);
-
-    /**
-     * Returns an integer representing the "complexity" of the generated plan.
-     * Currently, this should simply be the number of plans in the plan tree with this plan as the root (i.e. the
-     * number of descendants of this plan, including itself).
-     * @return the complexity of this plan
-     */
-    int getComplexity();
-
     /**
      * Returns the (zero or more) children of this plan.
      * @return the child plans
      */
     @Nonnull
     List<RecordQueryPlan> getChildren();
+
+    @Nonnull
+    @Override
+    default List<? extends QueryPlan<?>> getQueryPlanChildren() {
```

Comment:
I think we eventually want `getChildren()` to have the signature of `getQueryPlanChildren()` for the reasons that Mike described. I understand why we don't want to mess with relatively stable stuff while we write experimental code, though. I think this is fine for now but `getQueryPlanChildren()` should never really become a part of the stable API.

Reply:
In response to this comment and @MMcM's comment https://github.com/FoundationDB/fdb-record-layer/pull/404#issuecomment-465404235, I went with:

1. Marking `RecordQueryPlan::getChildren` as `UNSTABLE`.
1. Adding warnings explaining the status of the stability of `getChildren` and `QueryPlan::getQueryPlanChildren`.

I think that sums up how we want this API to evolve?

Reply:
Oh, and I think (as unfortunate as it sounds), we want `getChildren` to *not* return any child plans that aren't `RecordQueryPlan`s, which (in the case of a `ResolveRecordsPlan` I guess means returning an empty list). Our things should use `getQueryPlanChildren` internally and do their best not to break things.